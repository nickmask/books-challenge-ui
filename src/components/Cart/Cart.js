import React from "react";
import PropTypes from "prop-types";

import styles from "./Cart.module.css";

const Cart = ({ cartItems, totalValue, onRemoveCartItem }) => {
  const removeCartItem = event => {
    onRemoveCartItem(event.target.getAttribute("data-index"));
  };

  return (
    <div className={styles.container}>
      <h1 className={styles.pageHeader}>Cart</h1>
      <hr className={styles.separator} />
      {cartItems.length > 0 ? (
        <div>
          {cartItems.map((item, index) => (
            <div key={item.id + index}>
              <div className={styles.cartItem}>
                <img
                  className={styles.bookCover}
                  src={item.thumbnail}
                  alt={`Cover of ${item.title} book`}
                />
                <div className={styles.detailsContainer}>
                  <div className={styles.details}>
                    <h2 aria-label="Book title">{item.title}</h2>
                    <p aria-label="Book author">{`${item.author.firstName} ${
                      item.author.lastName
                    }`}</p>
                    <p aria-label="Book price">£{item.price}</p>
                  </div>
                  <div>
                    <button data-index={index} onClick={removeCartItem}>
                      Remove
                    </button>
                  </div>
                </div>
              </div>
              <hr className={styles.separator} />
            </div>
          ))}
          <div className={styles.buy}>
            <p>Total price: £{totalValue}</p>
            <div className={styles.buyButtonCotainer}>
              <span>
                <button
                  type="submit"
                  onClick={() => alert("To be implemented")}
                >
                  Buy all
                </button>
              </span>
            </div>
          </div>
        </div>
      ) : (
        <h2>Cart empty</h2>
      )}
    </div>
  );
};

Cart.proptype = {
  cartItems: PropTypes.object.isRequired,
  totalValue: PropTypes.string.isRequired,
  onRemoveCartItem: PropTypes.func.isRequired
};

export default Cart;
