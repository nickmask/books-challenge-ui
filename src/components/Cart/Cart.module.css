.container {
  padding: 0.5rem 1rem;
  margin: 0.5rem 0 0 0;
  background: #fff;
  box-shadow: 0.1rem 0.1rem 0.2rem rgba(0, 0, 0, 0.1);
}

.bookCover {
  object-fit: cover;
  width: 200px;
  height: 305px;
}

.cartItem {
  display: flex;
  flex-direction: row;
}

.details {
  display: flex;
  flex-direction: column;
  margin-left: 10px;
}

.detailsContainer {
  display: flex;
  justify-content: space-between;
  flex-grow: 1;
}

.separator {
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid #f0f0f0;
  margin: 0.5em 0;
  padding: 0;
}

.pageHeader {
  margin-bottom: 0;
}

.buy {
  display: flex;
  justify-content: space-between;
}

.buyButtonCotainer {
  display: flex;
  align-items: center;
}
