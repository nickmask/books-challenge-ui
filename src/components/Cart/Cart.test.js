import React from "react";
import { shallow } from "enzyme";

import Cart from "./Cart";

describe("Cart", () => {
  const cartItems = [
    {
      id: "c1dc42ab-548f-4f96-8350-2f56303d7532",
      author: {
        id: "b6545db7-2a50-4532-8225-3fec60470502",
        firstName: "John",
        lastName: "Doe"
      },
      price: 10,
      description:
        "Nulla sunt sunt pariatur officia et esse fugiat proident est. Ad laboris ullamco adipisicing nisi in ex. Magna voluptate culpa velit exercitation nisi ipsum est do.",
      stockAmount: 0,
      thumbnail:
        "https://images.unsplash.com/photo-1496484805162-42c00f79cdce?dpr=1&auto=format&fit=crop&w=568&h=379&q=60&cs=tinysrgb&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
    },
    {
      id: "c1dc42ab-548f-4f96-8350-2f56303d7532",
      author: {
        id: "b6545db7-2a50-4532-8225-3fec60470502",
        firstName: "John",
        lastName: "Doe"
      },
      price: 10,
      description:
        "Nulla sunt sunt pariatur officia et esse fugiat proident est. Ad laboris ullamco adipisicing nisi in ex. Magna voluptate culpa velit exercitation nisi ipsum est do.",
      stockAmount: 0,
      thumbnail:
        "https://images.unsplash.com/photo-1496484805162-42c00f79cdce?dpr=1&auto=format&fit=crop&w=568&h=379&q=60&cs=tinysrgb&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
    }
  ];

  it("should render correctly", () => {
    const component = shallow(<Cart cartItems={cartItems} />);

    expect(component).toMatchSnapshot();
  });

  it("should render the correct amount of cart items", () => {
    const component = shallow(<Cart cartItems={cartItems} />);

    const removeFromCartButton = component.find('button[children="Remove"]');

    expect(removeFromCartButton.length).toEqual(2);
  });
});
