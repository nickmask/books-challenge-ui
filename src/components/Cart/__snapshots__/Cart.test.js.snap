// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render correctly 1`] = `
<div
  className="container"
>
  <h1
    className="pageHeader"
  >
    Cart
  </h1>
  <hr
    className="separator"
  />
  <div>
    <div
      key="c1dc42ab-548f-4f96-8350-2f56303d75320"
    >
      <div
        className="cartItem"
      >
        <img
          alt="Cover of undefined book"
          className="bookCover"
          src="https://images.unsplash.com/photo-1496484805162-42c00f79cdce?dpr=1&auto=format&fit=crop&w=568&h=379&q=60&cs=tinysrgb&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
        />
        <div
          className="detailsContainer"
        >
          <div
            className="details"
          >
            <h2
              aria-label="Book title"
            />
            <p
              aria-label="Book author"
            >
              John Doe
            </p>
            <p
              aria-label="Book price"
            >
              £
              10
            </p>
          </div>
          <div>
            <button
              data-index={0}
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
      <hr
        className="separator"
      />
    </div>
    <div
      key="c1dc42ab-548f-4f96-8350-2f56303d75321"
    >
      <div
        className="cartItem"
      >
        <img
          alt="Cover of undefined book"
          className="bookCover"
          src="https://images.unsplash.com/photo-1496484805162-42c00f79cdce?dpr=1&auto=format&fit=crop&w=568&h=379&q=60&cs=tinysrgb&ixid=dW5zcGxhc2guY29tOzs7Ozs%3D"
        />
        <div
          className="detailsContainer"
        >
          <div
            className="details"
          >
            <h2
              aria-label="Book title"
            />
            <p
              aria-label="Book author"
            >
              John Doe
            </p>
            <p
              aria-label="Book price"
            >
              £
              10
            </p>
          </div>
          <div>
            <button
              data-index={1}
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
      <hr
        className="separator"
      />
    </div>
    <div
      className="buy"
    >
      <p>
        Total price: £
      </p>
      <div
        className="buyButtonCotainer"
      >
        <span>
          <button
            onClick={[Function]}
            type="submit"
          >
            Buy all
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
